3party/gtest/docu/docu_99gtest.md

```diff
@@ -0,0 +1,56 @@
+## Unit Testing: `gtest`
+
+Bricks constains a header-only port of Google's [`GoogleTest`](http://code.google.com/p/googletest/): a great C++ unit testing library open-sourced by [**Google**](https://www.google.com/finance?q=GOOG).
+
+A three-minute intro:
+
+1. **Logic**
+  
+  `ASSERT_*` interrupts the `TEST() { .. }` if failing,
+  
+   `EXPECT_*` considers the `TEST()` failed, but continues to execute it.
+
+2. **Debug Output**
+  
+   `ASSERT`-s and `EXPECT`-s can be used as output streams. No newline needed.
+  
+   `EXPECT_EQ(4, 2 * 2) << "Something is wrong with multiplication.";`
+
+3. **Conditions**
+  
+   `ASSERT`-s and `EXPECT`-s can use {`EQ`,`NE`,`LT`,`GT`,`LE`,`NE`,`TRUE`} after the underscore.
+  
+   This results in more meaningful human-readable test failure messages.
+
+4. **Parameters Order**
+  
+   For `{ASSERT,EXPECT}_{EQ,NE}`, put the expected value as the first parameter.
+  
+   For clean error messages wrt `expected` vs. `actual`.
+   
+
+5. **Exceptions**
+  
+   `ASSERT_THROW(statement, exception_type);` ensures the exception is thrown.
+
+6. **Death Tests**
+  
+   `ASSERT_DEATH(function(), "Expected regex for the last line of standard error.");`
+  
+   can be used to ensure certain call fails. The convention is to use the `"DeathTest"` suffix for those tests and to not mix functional tests with death tests.
+
+7. **Templated Tests**
+  
+   `gtest` supports templated tests, where objects of various tests are passed to the same test method.
+  
+   Each type results in the whole new statically compiled test.
```

Comment:
We do, but it still looks like too much trouble to get those involved/linked from here.